helm/nginx-ingress-controller-app/values.yaml

```diff
@@ -236,38 +236,31 @@ controller:
   service:
 
     # controller.service.enabled
-    # If true, create NodePort Service. Dynamically calculated during cluster creation.
+    # If true, create LoadBalancer Service. Dynamically calculated during cluster creation.
     # Do not overwrite this value.
-    enabled: false
+    enabled: true
 
     # controller.service.externalTrafficPolicy
     # Configures kube-proxy, denotes if this Service desires to have external traffic routed to node-local or cluster-wide endpoints
     #   Local - kube-proxy only proxies requests to local endpoints (does not forward traffic to other nodes), source IP preserved
     #   Cluster - kube-proxy proxies requests randomly across all endpoints (forwards traffic to other nodes if necessary), source IP NAT'd
     externalTrafficPolicy: "Local"
 
-    # controller.service.type
-    # Applies only to clusters running on AWS.
-    # Valid values: external, internal
-    type: external
-
-    # controller.service.nodePorts
-    # Sets the NodePorts that maps to the Ingress' ports 80 (http) and 443 (https).
-    # nodePort service type is not created for AWS provider as LoadBalancer is used there.
-    # Do not overwrite these values.
-    nodePorts:
-
-      # controller.service.nodePorts.http
-      http: 30010
+    # controller.service.public
+    # Applies to clusters running on AWS or Azure.
+    # Valid values: true for public, false for internal
+    public: true
 
-      # controller.service.nodePorts.https
-      https: 30011
+    # controller.service.type
+    # Default to LoadBalancer, support NodePort for testing purposes only.
```

Comment:
But we're running this onprem as well. And there nodeport is the only option

Reply:
NodePort for kvm is still managed by kvm-operator/k8scloudconfig/ignition. Eventually we may end up merging that management in nginx IC App. kvm/on-prem is not a priority for now.

We got to Azure in the first place because of external traffic policy configurability, public/internal support as well as trying to make nginx optional for Azure without downtime.

AWS legacy is also not priority. AWS node pools is already covered.